### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getAdapter' of null

### ISSUE THREAD

**Issue by benhamill** [mulesoft/api-workbench#133]
1. Type the following into a buffer named `test.raml`:
  ```
  #%RAML 1.0
  title: Test of Array+Union Types
  
  types:
    ItsComplicated:
      properties:
        []: string | string[]
      example: |
        {
          "bar": "baz"
        }
  ```
2. Look to the right! A scary red box.
3. Also: I think checking doesn't work any more after the box appears. If you change `"baz"` to `4`, the bottom of the window still says ":white_check_mark: No Issues".

**Atom Version**: 1.5.3
**System**: Mac OS X 10.10.5
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.20

### Stack Trace

Uncaught TypeError: Cannot read property 'getAdapter' of null

```
At /Users/bhamill/.atom/packages/api-workbench/main.js:10

TypeError: Cannot read property 'getAdapter' of null
    at e.validateFacets (/Users/bhamill/.atom/packages/api-workbench/main.js:10:5766)
    at e.validateArray (/Users/bhamill/.atom/packages/api-workbench/main.js:10:6905)
    at e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:10:2909)
    at e.validatePropValue (/Users/bhamill/.atom/packages/api-workbench/main.js:10:6223)
    at /Users/bhamill/.atom/packages/api-workbench/main.js:10:5453
    at Array.forEach (native)
    at /Users/bhamill/.atom/packages/api-workbench/main.js:10:5424
    at Array.forEach (native)
    at e.validateClass (/Users/bhamill/.atom/packages/api-workbench/main.js:10:5320)
    at e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:10:3064)
    at e.validatePropValue (/Users/bhamill/.atom/packages/api-workbench/main.js:10:6223)
    at /Users/bhamill/.atom/packages/api-workbench/main.js:10:5014
    at Array.forEach (native)
    at e.validateClass (/Users/bhamill/.atom/packages/api-workbench/main.js:10:4798)
    at e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:10:3064)
    at Object.r.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:11:4094)
    at e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:11:1365)
    at e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:10:10222)
    at Object.l [as validate] (/Users/bhamill/.atom/packages/api-workbench/main.js:9:27572)
    at t.e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:3:25552)
    at /Users/bhamill/.atom/packages/api-workbench/main.js:9:27456
    at Array.forEach (native)
    at p (/Users/bhamill/.atom/packages/api-workbench/main.js:9:27386)
    at Object.l [as validate] (/Users/bhamill/.atom/packages/api-workbench/main.js:9:27780)
    at t.e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:3:25552)
    at /Users/bhamill/.atom/packages/api-workbench/main.js:9:27456
    at Array.forEach (native)
    at p (/Users/bhamill/.atom/packages/api-workbench/main.js:9:27386)
    at Object.l [as validate] (/Users/bhamill/.atom/packages/api-workbench/main.js:9:27780)
    at t.e.validate (/Users/bhamill/.atom/packages/api-workbench/main.js:3:25552)
    at l (/Users/bhamill/.atom/packages/api-workbench/main.js:46:24427)
    at p (/Users/bhamill/.atom/packages/api-workbench/main.js:46:23995)
    at u (/Users/bhamill/.atom/packages/api-workbench/main.js:46:23443)
    at v (/Users/bhamill/.atom/packages/api-workbench/main.js:46:24722)
    at Workspace.module.exports.Workspace.observeTextEditors (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:290:9)
    at Object.n [as initEditorObservers] (/Users/bhamill/.atom/packages/api-workbench/main.js:46:23165)
    at Object.b [as consumeLinter] (/Users/bhamill/.atom/packages/api-workbench/main.js:1:1257)
    at Provider.module.exports.Provider.provide (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/service-hub/lib/provider.js:30:52)
    at /Applications/Atom.app/Contents/Resources/app.asar/node_modules/service-hub/lib/service-hub.js:55:26
    at doNTCallback0 (node.js:419:9)
    at process._tickCallback (node.js:348:13)
```

### Config

```json
{
  "core": {
    "projectHome": "~/src"
  },
  "api-workbench": {
    "openConsoleInSplitPane": false
  }
}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.20
linter, v1.11.3

# Dev
No dev packages
```

**Comment by ddenisenko** [comment]
Thank you, it is reproducible. It looks like "[]" part is causing the explosion.

**Comment by benhamill** [comment]
Does this being closed mean a fix was merged?

**Comment by sichvoge** [comment]
Seems it is fixed; your example works for me using 0.8.24
